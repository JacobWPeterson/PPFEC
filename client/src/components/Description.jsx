import React from 'react';
import styled from 'styled-components';

const DescWrapper = styled.div`
  border-right: 3px solid #aeaeae;
  flex-basis: 60%
`

const Slogan = styled.h1`
  color: #424242;
  font-weight: bold;
  font-size: 20px;
`

const Body = styled.p`
  color: #424242;
  font-weight: thin;
`

const Description = ({info}) => {
  return (
    <DescWrapper>
      <Slogan>{info.slogan}</Slogan>
      <Body>{info.description}</Body>
    </DescWrapper>
  )
}

export default Description;