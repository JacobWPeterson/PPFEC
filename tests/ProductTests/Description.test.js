/* eslint-disable no-undef */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount } from 'enzyme';

import Description from '../../client/src/components/ProductSubs/Description.jsx';

const props = {
  info: { slogan: 'Faster than a just about anything', description: 'The Pumped Up serves up crisp court style with a modern look. These shoes show off tennis-whites shades and are constructed with a supple leather upper and a classic rubber cupsole.' },
};

describe('Description', () => {
  const DescriptionComponent = mount(<Description {...props} debug />);
  const Slogan = DescriptionComponent.find('h1').text();
  const DescriptionText = DescriptionComponent.find('p').text();

  it('Should render correctly', () => {
    expect(DescriptionComponent).toMatchSnapshot();
  });

  it('renders the slogan', () => {
    expect(Slogan).toEqual(props.info.slogan);
  });

  it('renders the description', () => {
    expect(DescriptionText).toEqual(props.info.description);
  });
});
