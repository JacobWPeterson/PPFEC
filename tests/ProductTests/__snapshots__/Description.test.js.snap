// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description Should render correctly 1`] = `
<Description
  debug={true}
  info={
    Object {
      "description": "The Pumped Up serves up crisp court style with a modern look. These shoes show off tennis-whites shades and are constructed with a supple leather upper and a classic rubber cupsole.",
      "slogan": "Faster than a just about anything",
    }
  }
>
  <Description__DescWrapper>
    <div
      className="Description__DescWrapper-sc-1a8vwln-0 iuTlrA"
    >
      <Description__Slogan>
        <h1
          className="Description__Slogan-sc-1a8vwln-1 ixJlAJ"
        >
          Faster than a just about anything
        </h1>
      </Description__Slogan>
      <Description__Body>
        <p
          className="Description__Body-sc-1a8vwln-2 fDTlPa"
        >
          The Pumped Up serves up crisp court style with a modern look. These shoes show off tennis-whites shades and are constructed with a supple leather upper and a classic rubber cupsole.
        </p>
      </Description__Body>
    </div>
  </Description__DescWrapper>
</Description>
`;
